src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/RespondentTellSomethingElseService.java

```diff
@@ -0,0 +1,312 @@
+        + "event by %s.";
+    private static final String RES_TSE_FILE_NAME = "resTse.pdf";
+
+    private static final String TABLE_COLUMNS_MARKDOWN =
+        "| No | Application type | Applicant | Application date | Response due | Number of responses | Status |\r\n"
+            + "|:---------|:---------|:---------|:---------|:---------|:---------|:---------|\r\n"
+            + "%s\r\n";
+
+    private static final String TABLE_ROW_MARKDOWN = "|%s|%s|%s|%s|%s|%s|%s|\r\n";
+
+    /**
+     * Validate Give Details (free text box) or file upload is mandatory.
+     * @param caseData in which the case details are extracted from
+     * @return errors Error message
+     */
+    public List<String> validateGiveDetails(CaseData caseData) {
+        List<String> errors = new ArrayList<>();
+        RespondentTSEApplicationTypeData selectedAppData =
+                RespondentTellSomethingElseHelper.getSelectedApplicationType(caseData);
+        if (selectedAppData == null
+                || selectedAppData.getResTseDocument() == null && isNullOrEmpty(selectedAppData.getSelectedTextBox())) {
+            errors.add(GIVE_DETAIL_MISSING);
+        }
+        return errors;
+    }
+
+    /**
+     * Uses {@link EmailService} to generate an email to Respondent.
+     * Uses {@link UserService} to get Respondents email address.
+     * @param caseDetails in which the case details are extracted from
+     * @param userToken jwt used for authorization
+     */
+    public void sendAcknowledgeEmailAndGeneratePdf(CaseDetails caseDetails, String userToken) {
+        CaseData caseData = caseDetails.getCaseData();
+
+        if (ORDER_A_WITNESS_TO_ATTEND_TO_GIVE_EVIDENCE.equals(caseData.getResTseSelectApplication())) {
+            // No need to send email for Group C
+            return;
+        }
+
+        String email = userService.getUserDetails(userToken).getEmail();
+
+        if (NO.equals(caseData.getResTseCopyToOtherPartyYesOrNo())) {
+            emailService.sendEmail(emailTemplateId, email, buildPersonalisation(caseDetails, RULE92_ANSWERED_NO));
+            return;
+        }
+
+        String customisedText;
+
+        if (GROUP_B_TYPES.contains(caseData.getResTseSelectApplication())) {
+            customisedText = String.format(RULE92_ANSWERED_YES_GROUP_B, caseData.getResTseSelectApplication());
+        } else {
+            customisedText = String.format(RULE92_ANSWERED_YES_GROUP_A, caseData.getResTseSelectApplication());
+        }
+
+        emailService.sendEmail(emailTemplateId, email, buildPersonalisation(caseDetails, customisedText));
+    }
+
+    public void sendClaimantEmail(CaseDetails caseDetails) {
+        CaseData caseData = caseDetails.getCaseData();
+
+        if (ORDER_A_WITNESS_TO_ATTEND_TO_GIVE_EVIDENCE.equals(caseData.getResTseSelectApplication())) {
+            return;
+        }
+
+        if (NO.equals(caseData.getResTseCopyToOtherPartyYesOrNo())) {

```

Comment:
this and the above if statement could be combined if the return is the same

Reply:
fixed